tests/test_search.py

```diff
@@ -1,60 +1,201 @@
+            "/v1/search",
+            json_request_body=smartseq2_paired_ends_query,
+            expected_code=requests.codes.ok)
+        search_response = response.json
+        self.assertEqual(len(search_response), 2)
+        self.assertDictEqual(search_response['query'], smartseq2_paired_ends_query)
+        self.assertEqual(len(search_response['results']), 1)
+        self.assertEqual(search_response['results'][0]['bundle_id'], bundle_id)
+        self.assertEqual(search_response['results'][0]['bundle_url'], bundle_url)
+
+    def test_search_returns_no_results_when_no_documents_indexed(self):
+        self.verify_search_results(smartseq2_paired_ends_query)
+
+    def test_search_returns_no_result_when_query_does_not_match_indexed_documents(self):
         query = \
             {
                 "query": {
-                    "bool": {
-                        "must": [{
-                            "match": {
-                                "files.sample_json.donor.species": "Homo sapiens"
-                            }
-                        }, {
-                            "match": {
-                                "files.assay_json.single_cell.method": "Fluidigm C1"
-                            }
-                        }, {
-                            "match": {
-                                "files.sample_json.ncbi_biosample": "SAMN04303778"
-                            }
-                        }]
+                    "match": {
+                        "files.sample_json.donor.species": "xxx"
                     }
                 }
             }
-        self.assertPostResponse(
-            '/v1/search',
-            json_request_body=(query),
-            expected_code=requests.codes.ok)
+
+        self.populate_search_index(self.index_document, 1)
+        self.verify_search_results(query)
+
+    def test_search_returns_error_when_invalid_query_used(self):
+        # Some types of invalid queries are detected by Elasticsearch DSL
+        # and others by Elasticsearch itself, and the response codes differ.
+        invalid_query_data = [
+            (
+                {
+                    "query": {
+                        "mtch": {
+                            "SomethingInvalid": "xxx"
+                        }
+                    }
+                },
+                requests.codes.bad_request
+            ),
+            (
+                {
+                    "query": {
+                        "match": ["SomethingInvalid", "xxx"]
+                    }
+                },
+                requests.codes.internal_server_error
+            )
+        ]
+
+        self.populate_search_index(self.index_document, 1)
+        for query_data in invalid_query_data:
+            with self.subTest("Invalid Queries", bad_query=query_data):
+                self.assertPostResponse(
+                    "/v1/search",
+                    json_request_body=(query_data[0]),
```

Comment:
so if you do the `subTest` thing, i think you should be using `bad_query` inside the with statement.

or you could just use the for loop and no with statement.

Reply:
Is this what you mean?
```python
       for bad_query in invalid_query_data:
            with self.subTest("Invalid Queries", bad_query=bad_query):
                self.assertPostResponse(
                    "/v1/search",
                    json_request_body=(bad_query[0]),
                    expected_code=bad_query[1])
```

Reply:
Well, you could do:
```
for query_data in invalid_query_data:
    with self.subTest("Invalid Queries", bad_query=query_data):
        self.assertPostResponse(
           "/v1/search",
           json_request_body=(bad_query[0]),
           expected_code=bad_query[1])
```

or you could just do:

```
for bad_query in invalid_query_data:
    self.assertPostResponse(
       "/v1/search",
       json_request_body=(bad_query[0]),
       expected_code=bad_query[1])
```

Reply:
The first suggestion does not add bad_query into the scope. I also don't want to removed subtest, because that will ensure that all tests run if some fail.  I'll modified the subtest description and variable names. Other than that I don't see any benefit in modifying it further.

Reply:
Ohh, I did not realize that.  Well, that's a bummer.